hack/tools/tilt-prepare/main.go

```diff
@@ -470,16 +515,53 @@ func writeIfChanged(prefix string, path string, yaml []byte) error {
 	return nil
 }
 
-// prepareDeploymentForDebug alter controller deployments for working nicely with delve debugger;
-// most specifically, liveness and readiness probes are dropper and leader election turned off.
-func prepareDeploymentForDebug(prefix string, objs []unstructured.Unstructured) error {
+// prepareDeploymentCommandAndArgs sets the default Command and Args for the manager container.
+// extraArgs comes from the extra_args field in the tilt-settings file and are appended to the pre-existing ones.
+// This has the affect that if there are args already set in the manager Deployment the given ones via extra_args will
+// take precedence.
+func prepareDeploymentCommandAndArgs(prefix string, objs []unstructured.Unstructured, extraArgs extraArgs) error {
```

Comment:
Should the call to `prepareDeploymentForDebug` and `prepareDeploymentForObservability` move inside this new func?

Reply:
In my opinion we're at a point were it would be better if we have just one func which modifies the Deployment (but let's keep the updateDeployment util)

Reply:
I put everything into one func.